setup.py

```diff
@@ -132,6 +137,26 @@ def no_cythonize(extensions):
     extensions = cythonize_extensions(raw_extensions)
 else:
     extensions = no_cythonize(raw_extensions)
+    
+    
+# to compile or not to compile... depends if compiler is present:
+compiler = distutils.ccompiler.new_compiler()
+assert isinstance(compiler, distutils.ccompiler.CCompiler)
+distutils.sysconfig.customize_compiler(compiler)
+try:
+    compiler.compile([os.path.join(setup_path, 'test_compiler.c')])
+    os.remove([os.path.join(setup_path, 'test_compiler.o'))
+except CompileError:
+    warnings.warn("""WARNING: C compiler can't be found.
+Only slow pure python alternative functions will be available.
+To use fast implementation of some functions writen in cython/c either:
+a) check that you have compiler (EXACTLY SAME as your python
+distribution was compiled with) installed,
+b) use binary distribution of hyperspy (i.e. wheels, egg, (only osx and win)).
+Installation will continue in 10 sec...""")
+    extensions = []
+    from time import sleep
+    sleep(10) #wait 10 secs for user to notice the message
```

Comment:
if terminal used have some limit (often 1000 lines) it "flies" away. this will wait for 10 sec only if compiler is missing. if it is here, no waiting.